fix syntax error in single_cell/CITEseq/subcluster/function.R

dtm2451 opened the pull request:
## Original Scope
Per catch from @looc27 
adds 1 comma and removes another to fix a syntax error in the function definition

## Expanded Scope
- Adds option to batch correct ADT with harmony, instead of only allowing for Seurat's IntegrateData path
  - Controlled with a new `adt_batch_correction_method` input

dtm2451 committed:
fix syntax error

adds 1 comma, and removes another, to fix a syntax error in the function definition

dtm2451 committed:
CITEseq subset function - add harmony option for ADT batch correction

dtm2451 commented:
@erflynn and @ravipatel4, also adding you two as reviewers in case you have time to double-check the rpca / Seurat integration side of this code which is adapted from code that I'd originally received from one of you!

erflynn commented:
this is great!
@dtm2451  -- Is RunHarmony is running on the adt.pca instead of the PCA that might be already in the object (line 100)? I often DietSeurat for subsetting (I see you do this for the RPCA) to make sure I don't accidentally use existing dimreducs (an error I've made multiple times).

minor comments:
- I might move the find isotype controls outside the if/else (because done for both), and add an option to provide isotype control features as an argument
- Instead of having it be harmony or default to RPCA, you could also have an option to do "none" (see next note below)
- For the RNA only part -- I'd move that up, and only run the RNA only and not the ADT parts if that's requested? It's often the case that for subclustering for tissue data, we only look at the RNA and don't bother to recluster the ADT, skipping that would be helpful to save time -- also in that case, I don't necessarily want to redo batch correction on the ADT, especially since the ADT signal can be limited in a non-immune cluster so batch estimation is harder. Just a thought -- totally up to you and your use cases!

dtm2451 committed:
CITEseq subset function - documentation and fixes after feedback

dtm2451 commented:
> Is RunHarmony is running on the adt.pca instead of the PCA that might be already in the object (line 100)? I often DietSeurat for subsetting (I see you do this for the RPCA) to make sure I don't accidentally use existing dimreducs (an error I've made multiple times).

Ooof, good catch!  @looc27 you'll need to re-pull and re-run because, even further than this, I forgot to point the ADT harmony run on the 'adt.pca' dimred at all... so it currently targets the default 'pca' dimred, which is the RNA-side!  I'd bet that means you'd see errors in the WNN steps if you already tried running?
- [x] now runs `DietSeurat()` from the start, after setting RNA is the DefaultAssay
- [x] explicitly give `reduction` in all `RunHarmony()` calls

> minor comments:
> 
> * I might move the find isotype controls outside the if/else (because done for both), and add an option to provide isotype control features as an argument

- [x] exposed `adt_features` input, which defaults to non-"isotype"-named features.
  - [x] used for harmony path
  - [x] used for integration path (both per-library and the altogether post-integration)

> * Instead of having it be harmony or default to RPCA, you could also have an option to do "none" (see next note below)
> * For the RNA only part -- I'd move that up, and only run the RNA only and not the ADT parts if that's requested? It's often the case that for subclustering for tissue data, we only look at the RNA and don't bother to recluster the ADT, skipping that would be helpful to save time -- also in that case, I don't necessarily want to redo batch correction on the ADT, especially since the ADT signal can be limited in a non-immune cluster so batch estimation is harder. Just a thought -- totally up to you and your use cases!

- [x] added "none" option
- [x] added early error-point if given "none" here with `rna_only_umap_and_clustering = FALSE` because then there is no umap or clustering run
- [x] Skip WNN path when "none" used

I've also added some extra details in the documentation section and re-organized the inputs a bit to hopefully make things easier to understand!

@erflynn would you potentially have a chance to see if this function would now recreate some of your own re/sub-clustering faithfully?  If there are any other features needed, I'd like to added them!

dtm2451 committed:
CITEseq subset function - documentation updates
